### PR TITLE
Implement golden file based tests

### LINKED ISSUE #17: Golden file based tests
The ultimate goal of an operator is to generate and apply a set of YAML manifests to the cluster. Authors should have high confidence that changes to their operator does not unintentionally modify the output.

### PR DESCRIPTION
**Pull request by johnsonj**
This change provides a validator that exercises a given input (CR) to a
declarative.Reconciler matches the expected output (manifest).

fixes #17

### CODE REVIEW

**Comment by johnsonj** [comment]
This PR has a dependency on #18

File: pkg/test/golden/validator.go [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+	}
+
+	actualTmp, err := writeTmp(actual)
+	if err != nil {
+		return errors.Wrapf(err, "write actual yaml to temp file failed")
+	}
+	t.Logf("Wrote actual to %s", actualTmp)
+
+	// pls to use unified diffs, kthxbai?
+	cmd := exec.Command("diff", "-u", expectedPath, actualTmp)
```
Comment (justinsb) [inline pkg/test/golden/validator.go]: I think k8s has a diff package, but I think it's a great idea to rely on diff instead as the output from the k8s diff is really awkward, and kops wrote our own diff package and it's a massive pain...

**Comment by justinsb** [comment]
Looks great - and super-happy to see the test failing as it shows the tests are working :+1: 

You do need to rebase though

/approve
/lgtm

**Comment by justinsb** [comment]
Thanks for rebasing!

/approve
/lgtm

**Pull request merged**